src/test/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/integration/SupplementaryRisk.kt

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    fun `access forbidden when no read scope`() {
+      webTestClient.get().uri("/risks/supplementary/crn/$crn")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access allowed when role and scope supplied`() {
+      webTestClient.get().uri("/risks/supplementary/crn/$crn")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf("read")))
+        .exchange()
+        .expectStatus().is5xxServerError
+    }
+  }
+
+  @Nested
+  @DisplayName("Get by Source Risk ID Security")
+  inner class GetBySourceSecurity {
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.get().uri("/risks/supplementary/$sourceType/$sourceId")
+        .header("Content-Type", "application/json")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.get().uri("/risks/supplementary/$sourceType/$sourceId")
+        .headers(setAuthorisation())
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when no read scope`() {
+      webTestClient.get().uri("/risks/supplementary/$sourceType/$sourceId")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access allowed when role and scope supplied`() {
+      webTestClient.get().uri("/risks/supplementary/$sourceType/$sourceId")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf("read")))
+        .exchange()
+        .expectStatus().is5xxServerError
+    }
+  }
+
+  @Nested
+  @DisplayName("Craete new supplementary risk")
+  inner class PostNewSupplementaryRisk {
+
+    val requestBody = SupplementaryRiskDto(source = Source.INTERVENTION_REFERRAL, sourceId = "1234", crn = crn)
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/risks/supplementary")
+        .header("Content-Type", "application/json")
+        .bodyValue(requestBody)
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/risks/supplementary")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation())
+        .bodyValue(requestBody)
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when no read scope`() {
+      webTestClient.post().uri("/risks/supplementary")
+        .header("Content-Type", "application/json")
+        .headers(setAuthorisation(roles = listOf("ROLE_RISK_SUMMARY"), scopes = listOf()))
+        .bodyValue(requestBody)
+        .exchange()
+        .expectStatus().isForbidden
```

Comment:
Do we have a happy path for `PostNewSupplementaryRisk`?

Reply:
Just added it

Reply:
👍